bpf/lib/nodeport.h

```diff
@@ -1403,6 +1406,34 @@ static __always_inline int dsr_set_opt4(struct __ctx_buff *ctx,
 
 	return 0;
 }
+static __always_inline int dsr_clear_opt4(struct __ctx_buff *ctx,
+                    struct iphdr *ip4, struct dsr_opt_v4 *opt)
+{
+    __u32 iph_old, iph_new;
+    __u16 tot_len = bpf_ntohs(ip4->tot_len) - sizeof(struct dsr_opt_v4);
+    __be32 sum;
+
+    if (ip4->protocol == IPPROTO_TCP) {
+        return 0;
+    }
+
+    iph_old = *(__u32 *)ip4;
+    ip4->ihl -= sizeof(struct dsr_opt_v4) >> 2;
+    ip4->tot_len = bpf_htons(tot_len);
+    iph_new = *(__u32 *)ip4;
+
+    sum = csum_diff(&iph_old, 4, &iph_new, 4, 0);
+    sum = csum_diff(opt, sizeof(struct dsr_opt_v4), NULL, 0, sum);
+   
+    if (ctx_adjust_hroom(ctx, -8, BPF_ADJ_ROOM_NET,
```

Comment:
please add a new constant with descriptive name for the -8 so it's easier to understand what is being adjusted here